### PR TITLE
add `hexString` closure that return hex from UIColor

### PR DESCRIPTION
**Pull request by MajkCajk**
#### Checklist

- [x] Updated CHANGELOG.md.
- [x] Added tests (if applicable)

### CODE REVIEW

File: ACKategories/UIColorExtensions.swift [hunk 1]
```diff
@@ -27,6 +27,21 @@ public extension UIColor {
...
+    
+    /**
+     Returns color as hex string (eg. '#ff00ff').
+     Monochrome check included (works for white/black/clear).
+     */
+    var hexString: String {
+        let components = cgColor.components
+        let isMonochrome = cgColor.colorSpace?.model == .monochrome
+        
+        let r: CGFloat = isMonochrome ? components![0] : components![0]
```
Comment (olejnjak) [inline ACKategories/UIColorExtensions.swift]: Not sure about this `!` it seems to work with e.g. HSB colors, but if I use `UIColor(patternImage:)`, it crashes here. Therefore I think that this property should therefore return `String?`

File: Tests/ColorTests.swift [hunk 2]
```diff
@@ -10,6 +10,27 @@ import XCTest
...
+        XCTAssertEqual("#000000", hexColor)
+        
+        let whiteColor = UIColor.white
+        hexColor = whiteColor.hexString
+        XCTAssertEqual("#FFFFFF", hexColor)
+        
+        let greenColor = UIColor.green
+        hexColor = greenColor.hexString
+        XCTAssertEqual("#00FF00", hexColor)
+    }
```
Comment (olejnjak) [inline Tests/ColorTests.swift]: I'd add more tests for this as I can imagine that some cases might not work as expected (e.g. as mentioned above). Ideally a test for every type of `init` (well maybe we can skip `UIColor(cgColor:)`, `UIColor(ciColor:)` and definitely `UIColor(coder:)` but I think that other inits might be interesting. Or maybe simply turning the return type to optional would be enought, so we are sure that this extension doesn't make apps crash.

Comment (LukasHromadnik) [reply Tests/ColorTests.swift]: I think that skipping some `init`s is not a good idea. We should be 100 % sure that our code is safe no matter which initializer of `UIColor` is used

File: ACKategories/UIColorExtensions.swift (hunk 1 above)
Comment (LukasHromadnik) [reply ACKategories/UIColorExtensions.swift]: I agree that we should get rid of those exclamation marks and that the return type should be `String?`. `CGColor` also have a property named `numberOfComponents`. Would it be possible to use this property in the code?

Comment (MajkCajk) [reply ACKategories/UIColorExtensions.swift]: I added `numberOfComponents` to check for colors that comes from `UIColor(patternImage:)` init, cause these colors comes with one component only(containing 0 or 1). So they cannot be trasformed to usual colors.

File: Tests/ColorTests.swift [hunk 3]
```diff
@@ -10,6 +10,44 @@ import XCTest
...
+        let image = UIColor.green.image()
+        let patternImageColor = UIColor(patternImage: image)
+        hexColor = patternImageColor.hexString
+        XCTAssertEqual(nil, hexColor)
+        
+        let white = UIColor(white: 1, alpha: 1)
+        hexColor = white.hexString
+        XCTAssertEqual("#FFFFFF", hexColor)
+
+        let rgb = UIColor(red: 0, green: 1, blue: 0, alpha: 1)
```
Comment (LukasHromadnik) [inline Tests/ColorTests.swift]: I would split the tests into smaller parts. Each "block" should be IMO a seperate test

Comment (MajkCajk) [reply Tests/ColorTests.swift]: Done :). Travis build failed cause one of the inits is avaliable from iOS 10, so i added also a check there.

**Comment by olejnjak** [comment]
@LukasHromadnik I think that we're OK with this now, aren't we?

**Pull request merged**